src/components/Attachments/AttachmentCarousel/index.js

```diff
@@ -23,24 +27,37 @@ const viewabilityConfig = {
     itemVisiblePercentThreshold: 95,
 };
 
-function AttachmentCarousel({report, reportActions, source, onNavigate}) {
+function AttachmentCarousel({report, reportActions, source, onNavigate, setDownloadButtonVisibility, translate}) {
     const scrollRef = useRef(null);
 
     const {windowWidth, isSmallScreenWidth} = useWindowDimensions();
 
-    const {attachments, initialPage, initialActiveSource, initialItem} = useMemo(() => extractAttachmentsFromReport(report, reportActions, source), [report, reportActions, source]);
+    const [containerWidth, setContainerWidth] = useState(0);
+    const [page, setPage] = useState(0);
+    const [attachments, setAttachments] = useState([]);
+    const [activeSource, setActiveSource] = useState(source);
+    const [shouldShowArrows, setShouldShowArrows, autoHideArrows, cancelAutoHideArrows] = useCarouselArrows();
 
     useEffect(() => {
-        // Update the parent modal's state with the source and name from the mapped attachments
-        if (!initialItem) return;
-        onNavigate(initialItem);
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [initialItem]);
+        const attachmentsFromReport = extractAttachmentsFromReport(report, reportActions);
 
-    const [containerWidth, setContainerWidth] = useState(0);
-    const [page, setPage] = useState(initialPage);
-    const [activeSource, setActiveSource] = useState(initialActiveSource);
-    const [shouldShowArrows, setShouldShowArrows, autoHideArrows, cancelAutoHideArrows] = useCarouselArrows();
+        const initialPage = _.findIndex(attachmentsFromReport, (a) => a.source === source);
+
+        // Dismiss the modal when deleting an attachment during its display in preview.
+        if (initialPage === -1 && _.find(attachments, (a) => a.source === source)) {
```

Comment:
Just looking at the code I am not able to make out why did couldn't go in a common method? What's the difference between `index.js` and `index.native.js`, could merge this into a common method for both the platforms?

Reply:
Hmm… I thought about this, but I can't create a valuable function.
We need 2 variables from those two lines `initialPage` and `isAttachmentDeletingInLivePreview`.

```javascript
// I think we need to use `_.find` twice to make the function readable, but as you comment before we are not sure if _.find for a lot of attachments could be expensive or not?
function isAttachmentDeletingInLivePreview(prevAttachments, nextAttachments, activeSource){
    return _.find(prevAttachments, (a) => a.source === activeSource) 
        && !_.find(nextAttachments, (a) => a.source === activeSource)
}

// i think it redundant and not need to be a function
function getInitialPage(attachments, source){
    return _.findIndex(attachments, (a) => a.source === source);
}

```

Do you have any suggestion about the desired function? @mananjadhav